### DISCUSSION TITLE
Re-send verification e-mail results in "Bad Request" error

### DISCUSSION THREAD

**Discussion by Avega-DanDie** [zitadel/zitadel#2741] [Q&A] [answered]
Hello Zitadel team,

as part of our sign-up mechanism, we'd like to give to user the opportunity to have the verification mail re-send to his account. 
But if we make a POST request to an URL like this: https://api.zitadel.ch/management/v1/users/<userId>/email/_resend_verification and the following body: {"userId":"<userId>"} it's returned with  400 Bad Request error. The message states {"code":9, "message":"User is not yet initialised (COMMAND-E3fbw)", "details":[{"@type":"type.googleapis.com/zitadel.v1.ErrorDetail", "id":"COMMAND-E3fbw", "message":"User is not yet initialised"}]} 
I hope, I get the message right and it is complaining, just because the user is not yet fully "onboarded". This is a bit, let's say, confusing, because probably most users who are in need of that function will be in a un-initialised state. 
But maybe I'm wrong with my assumption and would be happy for some information.
Thanks and best regards

Daniel

**Comment by livio-a** [comment, marked as answer]
Hi @Avega-DanDie 

Thanks for the question. And you're right about the message and the not fully onboarded user.
The method you mentioned is for users, which have already been onboarded and changed their email.
To make it easier for us to differ the cases (but obviously not for the users 😉 ), we created a separate method for your case: "/users/{user_id}/_resend_initialization"
By calling this endpoint, you're also able to change the email address if needed, e.g. if the user had a typo during registration.
If you don't want to change the address, just leave if blank and only pass the user_id.

Hope I was able to clarify that a bit.

**Reply by Avega-DanDie** [reply]
Hello @livio-a 
thanks so much for the clarification and the extremely fast solution. That's absolutely the functionality we needed. 
I absolutely appreciate your quick response and your support. 
Best wishes

Daniel

**Comment by sanjaypaudel35** [comment]
Hello zitadel team,

Could you please guide me on how to add user email on access token.

**Reply by hifabienne** [reply]
hei @sanjaypaudel35 I am sorry I do not really understand what you need. And can you please open up a new discussion, I think it has nothing to do with this one
